src/riscv/lib/src/jit/builder/instr_map.rs

```diff
@@ -0,0 +1,233 @@
+// SPDX-FileCopyrightText: 2025 TriliTech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+//! Mapping between instruction addresses and instruction information.
+//!
+//! This module provides data structures and utilities to maintain a mapping between
+//! instruction addresses and their associated information. It includes [`AddrMap`] for storing
+//! addresses, [`InstrMap`] for storing instruction information, and [`InstrMapBuilder`] for
+//! constructing these maps in a consistent manner.
+//!
+//! `InstrMap` is currently only used in tests and has not yet been integrated into the JIT.
+#![cfg(test)]
+
+use std::fmt::Debug;
+use std::ops::Index;
+use std::ops::IndexMut;
+
+use crate::machine_state::memory::Address;
+
+/// A unique identifier for an instruction in an [`InstrMap`] or the associated [`AddrMap`].
+#[derive(Debug, Clone, Copy)]
+pub(crate) struct InstrId(usize);
```

Comment:
This is more a style nit than anything else, so feel free to resolve: This module is `pub(crate)`, you could use `pub` to simplify this module a little bit. The real gain is minimal, though.